nixos/modules/virtualisation/oci-containers.nix

```diff
@@ -256,10 +256,13 @@ let
     requires = dependsOn;
     environment = proxy_env;
 
-    path =
-      if cfg.backend == "docker" then [ config.virtualisation.docker.package ]
-      else if cfg.backend == "podman" then [ config.virtualisation.podman.package ]
-      else throw "Unhandled backend: ${cfg.backend}";
+    path = let
+      engine =
+        if cfg.backend == "docker" then config.virtualisation.docker.package
+        else if cfg.backend == "podman" then config.virtualisation.podman.package
+        else throw "Unhandled backend: ${cfg.backend}";
+    in [ engine ]
+      ++ lib.optional (builtins.elem "zfs" config.boot.supportedFilesystems) config.boot.zfs.package;
```

Comment:
I'll rephase my comment:  _nothing_ should be added here, it should be done it the docker and podman modules so we don't introduce any potential differences between them and this module.